### PR TITLE
Failing test case for V1VerticalPodAutoscalerCrudTest

### PR DESCRIPTION
**Pull request by matthyx**
## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [x] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by matthyx** [comment]
@rohanKanojia I have identified this serialization issue today... will work on it in the following days.

**Comment by rohanKanojia** [comment]
Could you please create an issue also so that I can assign it to you ?

**Comment by matthyx** [comment]
> Could you please create an issue also so that I can assign it to you ?

https://github.com/fabric8io/kubernetes-client/issues/3776

**Comment by matthyx** [comment]
@rohanKanojia done. Should I add an entry to the changelog?

**Comment by rohanKanojia** [comment]
@matthyx : thanks, yes please.

**Comment by matthyx** [comment]
@rohanKanojia ready :)

File: extensions/verticalpodautoscaler/tests/src/test/java/io/fabric8/verticalpodautoscaler/test/crud/V1VerticalPodAutoscalerCrudTest.java [hunk 1]
```diff
@@ -72,7 +72,12 @@ void shouldLoadVerticalPodAutoscaler() {
         "    kind:       Deployment",
         "    name:       my-app",
         "  updatePolicy:",
-        "    updateMode: \"Auto\""
+        "    updateMode: \"Auto\"",
+        "  resourcePolicy:",
+        "    containerPolicies:",
+        "      - containerName: container",
+        "        controlledResources: [\"cpu\", \"memory\"]",
+        "        controlledValues: RequestsOnly"
```
Comment (rohanKanojia) [inline extensions/verticalpodautoscaler/tests/src/test/java/io/fabric8/verticalpodautoscaler/test/crud/V1VerticalPodAutoscalerCrudTest.java:80]: Could you please add an assert statement below to check whether resourceName is deserialized into VPA correctly?

Comment (rohanKanojia) [reply extensions/verticalpodautoscaler/tests/src/test/java/io/fabric8/verticalpodautoscaler/test/crud/V1VerticalPodAutoscalerCrudTest.java:80]: Umm, sorry. I meant assertion for `vpa.getSpec().getResourcePolicy().getContainerPolicies().get(0).getControlledResources()` (field for which this fix is)

Comment (matthyx) [reply extensions/verticalpodautoscaler/tests/src/test/java/io/fabric8/verticalpodautoscaler/test/crud/V1VerticalPodAutoscalerCrudTest.java:80]: Sorry for the delay, done!

**Comment by matthyx** [comment]
@rohanKanojia ready to review :)

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**